multiple_listeners/test_mixed.py

```diff
@@ -12,9 +10,104 @@
 
 class TestMixedListeners(tester.TempestaTest):
 
-    backends = tc.backends
-    clients = tc.clients
-    tempesta = tc.tempesta
+    backends = [
+        {
+            'id': 'nginx',
+            'type': 'nginx',
+            'port': '8000',
+            'status_uri': 'http://${server_ip}:8000/nginx_status',
+            'config': """
+                pid ${pid};
+                worker_processes  auto;
+                events {
+                    worker_connections   1024;
+                    use epoll;
+                }
+                http {
+                    keepalive_timeout ${server_keepalive_timeout};
+                    keepalive_requests ${server_keepalive_requests};
+                    sendfile         on;
+                    tcp_nopush       on;
+                    tcp_nodelay      on;
+                    open_file_cache max=1000;
+                    open_file_cache_valid 30s;
+                    open_file_cache_min_uses 2;
+                    open_file_cache_errors off;
+                    error_log /dev/null emerg;
+                    access_log off;
+                    server {
+                        listen        ${server_ip}:8000;
+                        location / {
+                            return 200;
+                        }
+                        location /nginx_status {
+                            stub_status on;
+                        }
+                    }
+                }
+            """,
+        }
+
+    ]
+
+    clients = [
+        {
+            'id': 'curl-h2-true',
+            'type': 'external',
+            'binary': 'curl',
+            'ssl': True,
+            'cmd_args': '-Ikf --http2 https://127.0.0.4:443/ '
+        },
+        {
+            'id': 'curl-h2-false',
+            'type': 'external',
+            'binary': 'curl',
+            'ssl': True,
+            'cmd_args': '-Ikf --http2 https://127.0.0.4:4433/ '
+        },
+        {
+            'id': 'curl-https-true',
+            'type': 'external',
+            'binary': 'curl',
+            'ssl': True,
+            'cmd_args': '-Ikf --http1.1 https://127.0.0.4:4433/ '
+        },
+        {
+            'id': 'curl-https-false',
+            'type': 'external',
+            'binary': 'curl',
+            'ssl': True,
+            'cmd_args': '-Ikf --http1.1 https://127.0.0.4:443/ '
+        },
+    ]
+
+    tempesta = {
+        'config': """
+
+            listen 127.0.0.4:443 proto=h2;
+            listen 127.0.0.4:4433 proto=https;
+
```

Comment:
With https://github.com/tempesta-tech/tempesta/pull/1643 https://github.com/tempesta-tech/tempesta/issues/1624 is fixed, but the test must be reworked. For Tempesta FW config:
````
listen 127.0.0.4:443 proto=h2;
listen 127.0.0.4:4433 proto=https;

srv_group default {
	server 127.0.0.1:8000;
}

vhost tempesta-cat {
	proxy_pass default;
}

tls_match_any_server_name;
tls_certificate /root/tempesta/etc/tfw-root.crt;
tls_certificate_key /root/tempesta/etc/tfw-root.key;

cache 0;
block_action attack reply;

http_chain {
	-> tempesta-cat;
}
```

The tests, which must obviously pass:
````
# curl -Ikf --http2 https://127.0.0.4:443/
HTTP/2 200 
date: Wed, 17 Aug 2022 23:59:45 GMT
content-type: text/plain
content-length: 0
via: 2.0 tempesta_fw (Tempesta FW pre-0.7.0)
server: Tempesta FW/pre-0.7.0
```
```
# curl -Ikf --http1.1 https://127.0.0.4:4433/
HTTP/1.1 200 OK
Date: Thu, 18 Aug 2022 00:24:01 GMT
Content-Type: text/plain
Content-Length: 0
via: 1.1 tempesta_fw (Tempesta FW pre-0.7.0)
Server: Tempesta FW/pre-0.7.0
```
This test must not pass anymore
```
# curl -Ikf --http1.1 https://127.0.0.4:443/
curl: (35) error:0A000460:SSL routines::reason(1120)
```
and check in `dmesg` for `[tempesta tls] Warning: [::ffff:127.0.0.1] ClientHello: cannot find matching ALPN for http/1.1`.

The last test is more complicated
```
# curl -Ikf --http2 https://127.0.0.4:4433/
HTTP/1.1 200 OK
Date: Thu, 18 Aug 2022 00:17:13 GMT
Content-Type: text/plain
Content-Length: 0
via: 1.1 tempesta_fw (Tempesta FW pre-0.7.0)
Server: Tempesta FW/pre-0.7.0
```
note that the response is HTTP/1 while we have requested HTTP/2. We still can't enforce curl to use HTTP/2 only even with `--http2-prior-knowledge` instead of `---http2`. I think we should leave the test with the comment that there is an ALPN negotiation (actually `curl` sends the both `h2` and `https` in ALPN) and use the new `h2` library to send strictly HTTP/2 request and make sure that it isn't accepted on the HTTPS socket.